### PR TITLE
[SQL] Add Managed Database Log Replay service cmdlets

### PR DESCRIPTION
**Pull request by milanbrkic-ms**
## Description

Created 4 new cmdlets to support Log Replay service for Managed Database.

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by MilanBrkicFON** [comment]
This change is depending on https://github.com/Azure/azure-powershell/pull/12782

**Comment by MilanBrkicFON** [comment]
Please review change, so I can address potential comments on time and merge as soon the PR I am depending on is merged. Thanks

**Comment by MilanBrkicFON** [comment]
@msJinLei why did you remove needs-review label? Per my previous comment, I would like to get this pr review.

**Comment by msJinLei** [comment]
> @msJinLei why did you remove needs-review label? Per my previous comment, I would like to get this pr review.

The PR has the compilation error https://dev.azure.com/azure-sdk/public/_build/results?buildId=514190&view=logs&j=2f7369d1-fb97-5262-a571-f4e9c996b7d1&t=c65099ae-0a9a-5c25-4721-3560ad0166d5&l=199

Please fix first

**Comment by MilanBrkicFON** [comment]
Compilation errors are fix in https://github.com/Azure/azure-powershell/pull/12802. After this PR is merged you will need to merge #12782 which fixes invalid paths in code and then I will rebase my changes to master and everything should be fine.

So if I may ask you to ignore build errors and check for the code change, so I could address comments you might have.

**Comment by MilanBrkicFON** [comment]
Can you help me resolve this security issue I am getting ? Thanks

image

**Comment by MilanBrkicFON** [comment]
@msJinLei can you please review? I am chasing Milestone and need to merge until 10th

**Comment by msJinLei** [comment]
> @msJinLei can you please review? I am chasing Milestone and need to merge until 10th

The PR is basically OK. Please resolve the conflicts.

File: src/Sql/Sql/help/Stop-AzSqlInstanceDatabaseLogReplay.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+{{ Fill PassThru Description }}
```
Comment (msJinLei) [inline src/Sql/Sql/help/Stop-AzSqlInstanceDatabaseLogReplay.md:118]: Please add the description here

File: src/Sql/Sql/help/Complete-AzSqlInstanceDatabaseLogReplay.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -LastBackupName
+{{ Fill LastBackupName Description }}
```
Comment (msJinLei) [inline src/Sql/Sql/help/Complete-AzSqlInstanceDatabaseLogReplay.md:89]: Please add the description here

File: src/Sql/Sql/help/Complete-AzSqlInstanceDatabaseLogReplay.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+{{ Fill PassThru Description }}
```
Comment (msJinLei) [inline src/Sql/Sql/help/Complete-AzSqlInstanceDatabaseLogReplay.md:119]: Please add the description here

File: src/Sql/Sql/help/Start-AzSqlInstanceDatabaseLogReplay.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -LastBackupName
+{{ Fill LastBackupName Description }}
```
Comment (msJinLei) [inline src/Sql/Sql/help/Start-AzSqlInstanceDatabaseLogReplay.md:133]: Please add the description here

File: src/Sql/Sql/help/Start-AzSqlInstanceDatabaseLogReplay.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+{{ Fill PassThru Description }}
```
Comment (msJinLei) [inline src/Sql/Sql/help/Start-AzSqlInstanceDatabaseLogReplay.md:163]: Please add the description here

File: src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        public string LastBackupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the instance database collation to use
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The collation of the instance database to use.")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("SQL_Latin1_General_CP1_CI_AS", "Latin1_General_100_CS_AS_SC")]
+        public string Collation { get; set; }
```
Comment (msJinLei) [inline src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs:70]: You provide argument completer here but the related item in help file is not generated. Please update cmdlet the help file.

**Review by msJinLei — CHANGES_REQUESTED** [review]
Find some additional issue of help file. Please update the help files.

File: src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs (hunk 6 above)
Comment (MilanBrkicFON) [reply src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs:70]: I am not sure what you mean by saying this. I followed other examples and they have same help files when it comes to Collation property

Comment (msJinLei) [reply src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs:70]: > I am not sure what you mean by saying this. I followed other examples and they have same help files when it comes to Collation property

Please refer to -Edition of the cmdlet https://docs.microsoft.com/en-us/powershell/module/az.sql/new-azsqldatabase?view=azps-4.6.1
The value of in argument completer will be generated as accepted values in help file.

Comment (msJinLei) [reply src/Sql/Sql/ManagedDatabase/Cmdlet/StartAzureSqlInstanceDatabaseLogReplay.cs:70]: > I am not sure what you mean by saying this. I followed other examples and they have same help files when it comes to Collation property

Please neglect my comments here. I may misunderstanding something. I will double confirm. Please fix the other help file issues

**Review by msJinLei — APPROVED** [review]

**Pull request merged**